packages/api/src/routes/medical/internal-patient.ts

```diff
@@ -52,10 +52,16 @@ router.post(
   "/update-all",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const patientIdsRaw = getFrom("query").optional("patientIds", req);
+    const patientIds = patientIdsRaw?.split(",") ?? [];
+
     const facilities = await getFacilities({ cxId });
     let failedUpdateCount = 0;
     for (const facility of facilities) {
-      const patients = await getPatients({ cxId, facilityId: facility.id });
+      const allPatients = await getPatients({ cxId, facilityId: facility.id });
+      const patients = patientIds.length
+        ? allPatients.filter(p => patientIds.includes(p.id))
+        : allPatients;
```

Comment:
nit: doing the filter in the query would be more efficient than in memory - especially for cxs with large volumes of patients

Reply:
oh yeah, 💯 

I just didn't want to change much given the time available, but I agree, considering this is something that's being resource intensive its worth doing right